docs/source/tutorials/configure_data.rst

```diff
@@ -415,3 +415,19 @@ The above case shows to rewrite all the components in data config. But sometime,
         2. dummy_dataset_dataroot:
             - user_script https://github.com/microsoft/Olive/blob/main/test/unit_test/test_data_root.py#L31
             - json_config https://github.com/microsoft/Olive/blob/main/test/unit_test/test_data_root.py#L107
+
+.. note::
+    The components will be called with the following arguments along with any additional keyword arguments provided in the config:
+        - load_dataset: `data_dir` (keyword, optional)
+        - pre_process_data: `dataset` (positional)
+        - post_process_data: `output` (positional)
+        - dataloader: `dataset` (positional)
+
+    The components can get access to the `user_script` and `script_dir` from the data config by adding them as named arguments (positional or keyword) to the component function.
```

Comment:
Could we distinguish the argument of `user_script` and `script_dir`(component level) with the ones in data_config level.

Reply:
It is a bit confused when I first see this comment(if I do not see the following `text_generation_huggingface_pre_process`), why we expose user_script in component level.

Reply:
do you mean the component has it's own user script and script dir? 
For this, I think it will become problematic in the future if the system is not local. We won't know what all to mount to the system.
So, I thought it's easy to use the same user script for the config and the components. 

Even now, our data configs are not fully compatible with aml and docker system since the user script and script dirs are not mounted at all. when we fix this, it will be even more problematic if components have their own user script.

Reply:
maybe we can make it clear by using a special argument name like `__data_config_user_script__` and `__data_config_script_dir__`?

Reply:
Might have to do something similar to https://github.com/microsoft/Olive/pull/409 in the future for user script and script dir too.

Reply:
rolled back this change as discussed offline.